### PR TITLE
Add TypeScript definition

### PR DESCRIPTION
**Pull request by fregante**
@sindresorhus pls cek

`tsc` passes, but does that mean anything?

### CODE REVIEW

**Comment by fregante** [comment]
For reference, here's a TypeScript issue about automatically determining the element from a compound selector (plain element selectors already work correctly!) https://github.com/Microsoft/TypeScript/issues/29037

File: index.d.ts [hunk 1]
```diff
@@ -0,0 +1,42 @@
+// https://github.com/Microsoft/TypeScript/blob/9d3707d671592d030386956c9ce39e539b8d0972/src/lib/dom.generated.d.ts#L10581
+
+type baseElements = Element | Element[] | NodeList | DocumentFragment;
+
+export interface SelectDom {
```
Comment (sindresorhus) [inline index.d.ts:6]: I think you want to "export default" this one, so it's imported when you import `select-dom`.

Relevant: https://github.com/Microsoft/TypeScript/issues/3792#issuecomment-303526468

File: index.d.ts [hunk 2]
```diff
@@ -0,0 +1,42 @@
+// https://github.com/Microsoft/TypeScript/blob/9d3707d671592d030386956c9ce39e539b8d0972/src/lib/dom.generated.d.ts#L10581
+
+type baseElements = Element | Element[] | NodeList | DocumentFragment;
```
Comment (sindresorhus) [inline index.d.ts]: `baseElements` => `BaseElements`

**Comment by sindresorhus** [comment]
You should write some type tests: https://github.com/sindresorhus/typescript-definition-style-guide#testing

**Comment by fregante** [comment]
Does tsd-check work in the browser though?

**Comment by sindresorhus** [comment]
@bfred-it No. Why would you need it to?

**Comment by fregante** [comment]
Because the other regular tests are run in electron. Select-dom calls `document.querySelector`

**Comment by sindresorhus** [comment]
@bfred-it `tsd-check` doesn't actually evaluate the file. It just runs the TS type checker on it.

**Comment by fregante** [comment]
@sindresorhus I'll add that in the future. Releasing this for now since it seems to be working on RGH already

**Pull request merged**